vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/model/DataSourceDeclarationTest.java

```diff
@@ -63,4 +62,16 @@ public void equalityTest() throws MalformedURLException {
 		assertFalse(dataSourceDeclaration1.equals(null)); // written like this for recording coverage properly
 	}
 
+	@Test
+	public void testdataSourceDecalarationtoString() throws MalformedURLException {
+		DataSource dataSource1 = new SparqlQueryResultDataSource(new URL("https://example.org/"), "var",
+				"?var wdt:P31 wd:Q5 .");
+		Predicate predicate1 = Expressions.makePredicate("p", 3);
+		DataSourceDeclaration dataSourceDeclaration1 = new DataSourceDeclarationImpl(predicate1, dataSource1);
+		DataSource dataSource2 = new SparqlQueryResultDataSource(new URL("https://example.org/"), "var",
+				"?var wdt:P31 wd:Q5 .");
+		Predicate predicate2 = Expressions.makePredicate("p", 3);
+		DataSourceDeclaration dataSourceDeclaration2 = new DataSourceDeclarationImpl(predicate2, dataSource2);
+		assertEquals(dataSourceDeclaration1.toString(), dataSourceDeclaration2.toString());
```

Comment:
this method does not really test toString(), but the constructors. You test that by creating two objects with the same parameters, you get the same (string representations of the )objects.

You should test what you expect to get from toString() instead, by comparing it to a String constant.

Reply:
ok, you are right, but I need first to know about datasource Declarations, maybe someone can explain them to me in the meeting.